web3.go

```diff
@@ -52,36 +53,50 @@ func WeiAsGwei(w *big.Int) string {
 	return new(big.Rat).SetFrac(w, weiPerGwei).FloatString(9)
 }
 
-func CallConstantFunction(ctx context.Context, client Client, myabi abi.ABI, address, functionName string, parameters ...interface{}) (interface{}, error) {
-	var out interface{}
-	switch myabi.Methods[functionName].Outputs[0].Type.T {
+func convertOutputParameter(t abi.Argument) interface{} {
+	switch t.Type.T {
 	case abi.BoolTy:
-		out = new(bool)
+		return new(bool)
 	case abi.UintTy, abi.IntTy:
-		out = new(big.Int)
+		return new(big.Int)
 	case abi.StringTy:
-		out = new(string)
+		return new(string)
 	case abi.AddressTy:
-		out = new(common.Address)
+		return new(common.Address)
 	case abi.BytesTy, abi.FixedBytesTy:
-		out = new([]byte)
+		return new([]byte)
 	default:
-		out = new(string)
+		return new(string)
 	}
+}
 
+func CallConstantFunction(ctx context.Context, client Client, myabi abi.ABI, address, functionName string, parameters ...interface{}) ([]interface{}, error) {
+	var out []interface{}
+	var ret []interface{}
+	for _, t := range myabi.Methods[functionName].Outputs {
+		out = append(out, convertOutputParameter(t))
+	}
 	input, err := myabi.Pack(functionName, convertParameters(myabi.Methods[functionName], parameters)...)
-
 	if err != nil {
 		return nil, err
 	}
-
 	toAddress := common.HexToAddress(address)
 	res, err := client.Call(ctx, CallMsg{Data: input, To: &toAddress})
-	err = myabi.Unpack(&out, functionName, res)
+	if len(myabi.Methods[functionName].Outputs) > 1 {
+		err = myabi.Unpack(&out, functionName, res)
+		for _, o := range out {
+			valueof := reflect.ValueOf(o)
+			ret = append(ret, valueof.Elem())
+		}
+	} else {
+		err = myabi.Unpack(&out[0], functionName, res)
+		ret = append(ret, out[0])
+
+	}
 	if err != nil {
 		return nil, err
 	}
-	return out, nil
+	return ret, nil
```

Comment:
Why is the reflection necessary?
Can the function return stay as `(interface{}, error)`, and then this block be simplified to:
```go
if err != nil {
  return nil, err
}
if len(myabi.Methods[functionName].Outputs) == 1 {
  return out[0], nil
} 
return out, nil
```
Either way, the error check should come first.

Reply:
I have to use reflect because in other case it would return pointers to variables, not the values (in case of tuples)